viadot/sources/customer_gauge.py

```diff
@@ -0,0 +1,250 @@
+import pandas as pd
+from typing import Any, Dict, Literal
+from datetime import datetime
+
+from viadot.utils import handle_api_response
+from viadot.config import local_config
+from viadot.exceptions import CredentialError, APIError
+
+from prefect.utilities import logging
+
+logger = logging.get_logger()
+
+
+class CustomerGauge:
+    API_URL = "https://api.eu.customergauge.com/v7/rest/sync/"
+
+    def __init__(
+        self,
+        endpoint: Literal["responses", "non-responses"] = None,
+        credentials: Dict[str, Any] = None,
+    ):
+        """
+        A class to connect and download data using Customer Gauge API.
+        Below is the documentation for each of this API's gateways:
+            Responses gateway https://support.customergauge.com/support/solutions/articles/5000875861-get-responses
+            Non-Responses gateway https://support.customergauge.com/support/solutions/articles/5000877703-get-non-responses
+
+        Args:
+            endpoint (Literal["responses", "non-responses"]): Indicate which endpoint to connect. Defaults to None.
+            credentials (Dict[str, Any], optional): Credentials to connect with API containing client_id, client_secret. Defaults to None.
+
+        Raises:
+            ValueError: If endpoint is not provided or incorect.
+            CredentialError: If credentials are not provided in local_config or directly as a parameter
+        """
+        self.endpoint = endpoint
+        if endpoint is not None:
+            if endpoint in ["responses", "non-responses"]:
+                self.url = f"{self.API_URL}{endpoint}"
+            else:
+                raise ValueError(
+                    "Incorrect endpoint name. Choose: 'responses' or 'non-responses'"
+                )
+        else:
+            raise ValueError(
+                "Provide endpoint name. Choose: 'responses' or 'non-responses'"
+            )
+
+        if credentials is not None:
+            self.credentials = credentials
```

Comment:
You can also add an option with local_config - import local config. Then you can replace `credentials=credentials` with `credentials= DEFAULT_CREDENTIALS or credentials`

Reply:
as in conversation above